library/IvozProvider/Gearmand/Invoices/Generator.php

```diff
@@ -163,7 +165,14 @@ protected function _getCallData(\IvozProvider\Model\Invoices $invoice)
             );
             $this->_fixedCostTotal  += number_format(ceil($subTotal*10000)/10000, 4);
         }
+
+        $dbAdapter = $invoice->getMapper()->getDbTable()->getAdapter();
+        $updateCallsInvoiceIdQuery = 'UPDATE `kam_acc_cdrs` SET `invoiceId` = ' . $invoice->getId();
+        $updateCallsInvoiceIdQuery .= ' WHERE id in (%s)';
```

Comment:
It's executed every loop, 100 calls actually. But sure, I can change it so that it's executed once everything is computed